src/Operators/spectralelement.jl

```diff
@@ -233,6 +233,53 @@ Base.@propagate_inbounds _resolve_operator_args(slabidx, arg, xargs...) = (
 
 
 # Functions for CUDASpectralStyle
+struct PlaceholderSpace <: Spaces.AbstractSpace end
+
+
+strip_space(obj, parent_space) = obj
+
+function strip_space(field::Field, parent_space)
+    current_space = axes(field)
+    if current_space == parent_space
+        new_space = PlaceholderSpace()
+    else
+        new_space = current_space
+    end
+    return Field(Fields.field_values(field), new_space)
+end
+
+function strip_space(
+    bc::Base.Broadcast.Broadcasted{Style},
+    parent_space,
+) where {Style}
+    current_space = axes(bc)
+    if current_space == parent_space
+        new_space = PlaceholderSpace()
+    else
+        new_space = current_space
+    end
+    return Base.Broadcast.Broadcasted{Style}(
+        bc.f,
+        map(arg -> strip_space(arg, current_space), bc.args),
```

Comment:
Does this need to be recursive?

Reply:
Probably, but it's not in a tight loop, so we're probably fine.

Reply:
Just so I understand correctly, does this mean that if a BC expression has 8 levels of recursion, then this would strip it down to 7?

Reply:
Ah, ok, got it.